kong/tools/http.lua

```diff
@@ -527,4 +527,35 @@ do
 end
 
 
+do
+  local replace_dashes = require("kong.tools.string").replace_dashes
+
+  function _M.get_header(name, ctx)
+    local value = ngx.var["http_" .. replace_dashes(name)]
+
+    if not value or not value:find(", ", 1, true) then
+      return value
+    end
+
+    local headers
+
+    if ctx then
+      if not ctx.headers then
```

Comment:
```suggestion
      if not ctx.cached_request_headers then
```

to avoid potential naming conflict. `ctx.headers` is an easy name and could easily be used by custom plugins.

Reply:
done.